chatui/src/app/modules/home/create-agent-dialog.tsx

```diff
@@ -7,15 +8,69 @@ import {
 	DialogHeader,
 	DialogTitle,
 } from '@memgpt/components/dialog';
-import { Input } from '@memgpt/components/input';
+import { DropdownMenu, DropdownMenuContent, DropdownMenuItem, DropdownMenuTrigger } from '@memgpt/components/dropdown-menu'
 import { Label } from '@memgpt/components/label';
 import { Loader2 } from 'lucide-react';
 import { useAgentsCreateMutation } from '../../libs/agents/use-agents.mutation';
 import { useAuthStoreState } from '../../libs/auth/auth.store';
+import { useState, useEffect } from 'react';
+import { useModelsQuery } from '../../libs/models/use-models.query';
+import { usePersonasQuery } from '../../libs/personas/use-personas.query';
+import { useHumansQuery } from '../../libs/humans/use-humans.query';
+import { useModelStore } from '../../libs/models/model.store';
+import { usePersonaStore } from '../../libs/personas/persona.store';
+import { useHumanStore } from '../../libs/humans/human.store';
+
+type DropdownValues = {
+    name: string[];
+    human: string[];
+    persona: string[];
+    model: string[];
+};
 
 const CreateAgentDialog = (props: { open: boolean; onOpenChange: (open: boolean) => void }) => {
 	const auth = useAuthStoreState();
 	const createAgent = useAgentsCreateMutation(auth.uuid);
+
+    const [dropdownValues, setDropdownValues] = useState<DropdownValues>({ 
+		name: [], 
+		human: [], 
+		persona: [], 
+		model: [] 
+	});
+
+    // Fetch models, humans, and personas using the custom hooks
+    const { data: modelsData } = useModelsQuery(auth.uuid);
+    const { data: humansData } = useHumansQuery(auth.uuid);
+    const { data: personasData } = usePersonasQuery(auth.uuid);
+
+    // Zustand stores to manage the state
+    const { setModels } = useModelStore();
+    const { setHumans } = useHumanStore();
+    const { setPersonas } = usePersonaStore();
+
+    // Effect to update the dropdown values when the data is fetched
+    useEffect(() => {
```

Comment:
Do you think we need to store this information globally in Zustand? I think just passing the query results to our selects is all we would need. 

TanStack Query already has the results available globally for us!

Reply:
Yes please! Agent is a little different. We keep track of the currently selected agent in a global zustand slice of state.